models/cice/ice_postprocessing_mod.f90

```diff
@@ -0,0 +1,968 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+!
+! $Id$
+
+module ice_postprocessing_mod
+
+use types_mod, only : r8
+use  utilities_mod, only : E_ERR, error_handler
+use  netcdf_utilities_mod, only : nc_check
+use netcdf
+
+implicit none
+
+! general variable initialization
+character(len=3)   :: nchar
+character(len=512) :: string1, string2, msgstring
+character(len=15)  :: varname
+integer :: n, i, j
+
+! -----------------------------------------------------------------------------
+contains
+! -----------------------------------------------------------------------------
+
+subroutine get_3d_variable(ncid, varname, var, filename)
```

Comment:
There routines in dart to read and write netcdf variables, no need to write your own.